cli/src/commands/tunnels.rs

```diff
@@ -228,10 +245,22 @@ async fn serve_with_csa(
 	let shutdown_tx = if let Some(tx) = shutdown_rx {
 		tx
 	} else {
-		let (tx, rx) = oneshot::channel();
+		let (tx, rx) = mpsc::channel::<ShutdownSignal>(2);
+		if let Some(process_id) = gateway_args.parent_process_id {
+			let tx = tx.clone();
+			info!(log, "checking for parent process {}", process_id);
+			tokio::spawn(async move {
+				let mut s = sysinfo::System::new();
+				let pid = Pid::from(process_id);
+				while s.refresh_process(pid) {
+					sleep(Duration::from_millis(500)).await;
```

Comment:
500ms might be a little aggressive, I'm not sure how intensive `refresh_process` is.

Reply:
I changed it to 2s